adapters/adagio/adagio.go

```diff
@@ -0,0 +1,141 @@
+package adagio
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+// Builder builds a new instance of the Adagio adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+type adapter struct {
+	endpoint string
+}
+
+type extBid struct {
+	Prebid *openrtb_ext.ExtBidPrebid
+}
+
+// MakeRequests prepares the HTTP requests which should be made to fetch bids.
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	json, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	if request.Device != nil {
+		if len(request.Device.IPv6) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IPv6)
+		}
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+	}
+
+	if request.Test == 0 {
+		// Gzip the body
+		// Note: Gzipping could be handled natively later: https://github.com/prebid/prebid-server/issues/1812
+		var bodyBuf bytes.Buffer
+		gz := gzip.NewWriter(&bodyBuf)
+		_, err = gz.Write(json)
+		if err == nil {
```

Comment:
Nitpick: You can make the error checking inline:
```
if _, err := gz.Write(json); err == nil {
  if err := gz.Close(); err == nil {
    json = bodyBuf.Bytes()
    headers.Add("Content-Encoding", "gzip")
    // /!\ Go already sets the `Accept-Encoding: gzip` header. Never add it manually, or Go won't decompress the response.
    //headers.Add("Accept-Encoding", "gzip")
  }
}
```

Reply:
Yep, I updated